server/punishments.js

```diff
@@ -287,17 +289,17 @@ Punishments.savePunishments = function () {
 		});
 
 		let buf = 'Punishment\tUser ID\tIPs and alts\tExpires\r\n';
-		saveTable.forEach((entry, id) => {
+		for (const [entry, id] of saveTable) {
 			buf += Punishments.renderEntry(entry, id);
-		});
+		}
 		return buf;
 	});
 };
 
 Punishments.saveRoomPunishments = function () {
 	FS(ROOM_PUNISHMENT_FILE).writeUpdate(() => {
 		const saveTable = new Map();
-		Punishments.roomIps.nestedForEach((/** @type {Punishment} */ punishment, /** @type {string} */ roomid, /** @type {string} */ ip) => {
+		Punishments.roomIps.nestedforEach((/** @type {Punishment} */ punishment, /** @type {string} */ roomid, /** @type {string} */ ip) => {
```

Comment:
```suggestion
		Punishments.roomIps.nestedForEach((/** @type {Punishment} */ punishment, /** @type {string} */ roomid, /** @type {string} */ ip) => {
```

Uppercase `F`; you also have this problem at every other instance of this.